### PR TITLE
Implement functions over bytes

### PR DESCRIPTION
**Pull request by FantomeBeignet**
This pull request implements the same functions as in the `Bigstring` and `Cstruct` modules over the native `bytes` type for convenience.

### CODE REVIEW

**Comment by dinosaure** [comment]
It's probably better to use `Eqaf.compare` and `Bytes.unsafe_to_string` to avoid a duplication of the code, WDYT?

**Comment by FantomeBeignet** [comment]
To be perfectly honest, I did not go this way because I felt like I did not know enough about the OCaml internals to guarantee this would still be constant time. After a re-read of the documentation and brief read through the compiler source, it seems this would be perfectly safe, so I'll update the PR.

File: lib/eqaf_bytes.ml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  let a' = Bytes.unsafe_to_string a in
+  let b' = Bytes.unsafe_to_string b in
+  Eqaf.compare_be_with_len ~len a' b'
+
+let compare_be a b =
+  let a' = Bytes.unsafe_to_string a in
+  let b' = Bytes.unsafe_to_string b in
+  Eqaf.compare_be a' b'
+
+let find_uint8 ?(off= 0) ~f b =
```
Comment (cfcs) [inline lib/eqaf_bytes.ml:26]: is `= 0` necessary here?

**Review by cfcs — CHANGES_REQUESTED** [review]
Only one minor comment, redefining the default for `?off` in `find_uint8` seems like a leftover artifact from debugging/development?
Otherwise looks good to me. WDYT @dinosaure

File: lib/eqaf_bytes.ml (hunk 1 above)
Comment (dinosaure) [reply lib/eqaf_bytes.ml:26]: You can indeed keep the optional argument and give it as is to `Eqaf.find_uint8` instead of specifying `0`

**Comment by dinosaure** [comment]
Yes, it looks good for me :+1: (I forgotten this PR for a while).

**Comment by dinosaure** [comment]
The error in the CI seems unrelated to this PR. I will merge and cut a release as soon as possible, thanks @FantomeBeignet for your work and thanks @cfcs for your review (and remind me to work on the projet :+1:).

**Pull request merged**